### PR TITLE
Fixes #587 form enter submit

### LINKED ISSUE #587: Form keyboard "enter" clears should submit

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @dtsnell4 @jbrundage @jchambers-ln @drealeed
 
Please Review 

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/GH-587-FORM-FIX/demos/marshaller/index.html?Hostname=10.173.147.1&Port=8010&Protocol=http&Wuid=W20150720-160015&ResultName=leeddx_issue_983_formvalueref_Comp_Ins026_DDL

**Comment by drealeed** [comment]
tested against this branch, working

**Comment by GordonSmith** [comment]
@mlzummo Can you imrpoive the commit message along the lines of: https://github.com/hpcc-systems/HPCC-Platform/wiki/Git-commit-message-guidelines (I use the commit message to create the "change log" entries: https://github.com/hpcc-systems/Visualization/releases)

Use GH-NNN rather than HPCC-NNN to reference a GitHub issue.

**Comment by GordonSmith** [comment]
Also I think simply changing the "type" from button to "submit" will achieve the same effect without having to mess around with the charCodes

**Comment by mlzummo** [comment]
i believe i tried that already to no effect, but i will look again

**Comment by mlzummo** [comment]
also we would have to use input type="submit" vs a html5  <button> attr?

**Comment by mlzummo** [comment]
@GordonSmith Please Re-Review

File: src/form/Input.js [hunk 1]
```diff
@@ -84,8 +84,7 @@
             w.blur(w);
         });
         var context = this;
-        this._inputElement.on("change", function (w) {
-
+        this._inputElement.on("keyup", function (w) {
```
Comment (mlzummo) [inline src/form/Input.js]: keep this ... change rest to on input type = submit

File: src/form/Form.css [hunk 2]
```diff
@@ -19,7 +19,7 @@
     vertical-align:middle;
 }
 
-.form_Form tfoot button{
+.form_Form tfoot button {
```
Comment (mlzummo) [inline src/form/Form.css]: need to add input type = submit to css

**Comment by mlzummo** [comment]
@GordonSmith Input type=submit still did not work, clear is selected but if i change the order of rendering of those buttons it will work .... example below with suggested changes

http://rawgit.com/mlzummo/Visualization/GH-587-FORM-FIX_2/demos/marshaller/index.html?Hostname=10.173.147.1&Port=8010&Protocol=http&Wuid=W20150720-160015&ResultName=leeddx_issue_983_formvalueref_Comp_Ins026_DDL

**Comment by mlzummo** [comment]
issue described here:

http://stackoverflow.com/questions/4763638/enter-triggers-button-click

**Comment by mlzummo** [comment]
http://stackoverflow.com/questions/1963245/multiple-submit-buttons-on-html-form-designate-one-button-as-default

**Comment by mlzummo** [comment]
if you want me to just change the order of the clear and submit button that will fix it too

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/Visualization/pull/635

**Pull request closed without merging**